doc/_extensions/zephyr/domain/templates/board-catalog.html

```diff
@@ -0,0 +1,59 @@
+{#
+  Copyright (c) 2024, The Linux Foundation.
+  SPDX-License-Identifier: Apache-2.0
+#}
+
+<form class="filter-form">
+  <input type="text" id="name" placeholder="Name" style="flex-basis: 100%" />
+  <div class="select-container">
+    <select id="arch">
+      <option value="">Select an architecture</option>
+      <option value="nios2">Altera Nios II</option>
+      <option value="arm">ARM</option>
+      <option value="arm64">ARM 64</option>
+      <option value="mips">MIPS</option>
+      <option value="posix">POSIX</option>
+      <option value="riscv">RISC-V</option>
+      <option value="sparc">SPARC</option>
+      <option value="arc">Synopsys DesignWare ARC</option>
+      <option value="x86">x86</option>
```

Comment:
I assume this an idea for a future PR, but this could probably be generated instead of hardcoded.

Reply:
Ya generating the unique list of archs is easy, but what's hardcoded here for convenience is the human readable version for each, as I don't think we have that anywhere. I guess people probably wouldn't mind if we'd use the short names though, as they're pretty self-explanatory and commonly used :)